nbgitautosync/static/index.js

```diff
@@ -0,0 +1,302 @@
+                that.eventSource.close();
+            }
+            that._emit(data.phase, data);
+        });
+        this.eventSource.addEventListener('error', function(error) {
+            console.log(arguments);
+            that._emit('error', error);
+        });
+    };
+
+    function GitSyncView(termSelector, progressSelector, termToggleSelector) {
+        // Class that encapsulates view rendering as much as possible
+        this.term = new Terminal({
+            convertEol: true,
+        });
+        this.term.open($(termSelector)[0]);
+        this.$progress = $(progressSelector);
+
+        this.$termToggle = $(termToggleSelector);
+
+        var that = this;
+        this.$termToggle.click(function() {
+            $(termSelector).parent().toggleClass('hidden');
+            that._fitTerm();
+        });
+    }
+
+
+    GitSyncView.prototype.setProgressValue = function(val) {
+        this.$progress.attr('aria-valuenow', val);
+        this.$progress.css('width', val + '%');
+    };
+
+    GitSyncView.prototype.getProgressValue = function() {
+        return parseFloat(this.$progress.attr('aria-valuenow'));
+    };
+
+    GitSyncView.prototype.setProgressText = function(text) {
+        this.$progress.children('span').text(text);
+    };
+
+    GitSyncView.prototype.getProgressText = function() {
+        return this.$progress.children('span').text();
+    };
+
+    GitSyncView.prototype.setProgressError = function(isError) {
+        if (isError) {
+            this.$progress.addClass('progress-bar-danger');
+        } else {
+            this.$progress.removeClass('progress-bar-danger');
+        }
+    };
+
+    GitSyncView.prototype._fitTerm = function() {
+        // Vendored in from the xterm.js fit addon
+        // Because I can't for the life of me get the addon to be
+        // actually included here as an require.js thing.
+        // And life is too short.
+        var term = this.term;
+        if (!term.element.parentElement) {
+            return null;
+        }
+        var parentElementStyle = window.getComputedStyle(term.element.parentElement),
+            parentElementHeight = parseInt(parentElementStyle.getPropertyValue('height')),
+            parentElementWidth = Math.max(0, parseInt(parentElementStyle.getPropertyValue('width')) - 17),
+            elementStyle = window.getComputedStyle(term.element),
+            elementPaddingVer = parseInt(elementStyle.getPropertyValue('padding-top')) + parseInt(elementStyle.getPropertyValue('padding-bottom')),
+            elementPaddingHor = parseInt(elementStyle.getPropertyValue('padding-right')) + parseInt(elementStyle.getPropertyValue('padding-left')),
+            availableHeight = parentElementHeight - elementPaddingVer,
+            availableWidth = parentElementWidth - elementPaddingHor,
+            container = term.rowContainer,
+            subjectRow = term.rowContainer.firstElementChild,
```

Comment:
I think `subjectRow` may still be able to be jquery-ified in the following lines w/ `html()`, `css()`, and `clientRect()`